adapters/Unity/UnityAdapter/GADMUnityInterstitialAd.m

```diff
@@ -209,4 +184,45 @@ - (void)unityAdsAdLoaded:(nonnull NSString *)placementId {
   }
 }
 
+#pragma mark - UnityAdsShowDelegate Methods
+
+- (void)unityAdsShowClick:(NSString *)placementId {
+  // TODO: unityAdsDidClick?
+  id<GADMAdNetworkConnector> strongNetworkConnector = _connector;
```

Comment:
why do we need to to the assignment here? 
_connector by itself still weak, strongNetworkConnector is strong for this scope.
but lets assume _connector is nullified?
calling like that technically wont cause any crash, since in obj we are allowed to send messages to NULL object
```
[_connector adapterDidGetAdClick:strongAdapter];
```
for the strong adapter I would assume its the same, but we need to check if it is  null or not before passing to the function.
However if there is a possibility that we have a background thread that can nullify the _adapter, because we hold it as a week reference, I can understand that we need to make sure that the reference is not deallocated ... hm this is an interesting topic. (talking to myself :) )

Reply:
This is an example of how it was in the old adapter when moving to use the show handlers.  https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Unity/UnityAdapter/GADMUnityInterstitialAd.m#L150

Maybe we can review if it is needed when we do the followup refactor/cleanup of the adapters?